x-pack/plugins/lists/server/routes/read_privileges_route.mock.ts

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+interface Cluster {
+  monitor_ml: boolean;
+  manage_ccr: boolean;
+  manage_index_templates: boolean;
+  monitor_watcher: boolean;
+  monitor_transform: boolean;
+  read_ilm: boolean;
+  manage_security: boolean;
+  manage_own_api_key: boolean;
+  manage_saml: boolean;
+  all: boolean;
+  manage_ilm: boolean;
+  manage_ingest_pipelines: boolean;
+  read_ccr: boolean;
+  manage_rollup: boolean;
+  monitor: boolean;
+  manage_watcher: boolean;
+  manage: boolean;
+  manage_transform: boolean;
+  manage_api_key: boolean;
+  manage_token: boolean;
+  manage_ml: boolean;
+  manage_pipeline: boolean;
+  monitor_rollup: boolean;
+  transport_client: boolean;
+  create_snapshot: boolean;
+}
+
+interface Index {
+  [indexName: string]: {
+    all: boolean;
+    manage_ilm: boolean;
+    read: boolean;
+    create_index: boolean;
+    read_cross_cluster: boolean;
+    index: boolean;
+    monitor: boolean;
+    delete: boolean;
+    manage: boolean;
+    delete_index: boolean;
+    create_doc: boolean;
+    view_index_metadata: boolean;
+    create: boolean;
+    manage_follow_index: boolean;
+    manage_leader_index: boolean;
+    write: boolean;
+  };
+}
+
+interface IndexPrivilege {
+  application: {};
+  cluster: Cluster;
+  has_all_requested: boolean;
+  index: Index;
+  username: string;
+}
+
+export interface Privilege {
```

Comment:
The types in this file seem redundant with [the ones in security_solution](https://github.com/elastic/kibana/blob/master/x-pack/plugins/security_solution/public/detections/containers/detection_engine/alerts/types.ts#L49-L100). 

Given that we're already using the `readPrivileges` function from security_solution, we should probably grab the types from the same place. 

Whichever set we choose, we should use them to refine the `unknowns` on the [frontend](https://github.com/elastic/kibana/blob/master/x-pack/plugins/lists/public/lists/api.ts#L201-L206) and [backend](https://github.com/elastic/kibana/blob/master/x-pack/plugins/security_solution/server/lib/detection_engine/privileges/read_privileges.ts#L9-L12).

Reply:
Yeah and neither is io-ts types and I have been a bit afraid to use strong types and validation from permissions with Elastic as I don't know how dynamic it is.

The ones you mention right now within security_solution are front end only at the moment. I could put those maybe into the common section, then expose them as a deps within lists and pull them in here. I kind of wish I had a universal type from core somewhere for Privilege or from elastic types.

The newer elastic client might have them, maybe? That would be better because if you look at the siem calls into elastic for permissions that's where it wants the types but uses an unknown there as we had to use the transport layer for this and did not have a first class API call for it:
https://github.com/elastic/kibana/blob/master/x-pack/plugins/security_solution/server/lib/detection_engine/privileges/read_privileges.ts#L12

The newer API at least has a non-transport version:
https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/api-reference.html#_security_hasprivileges

It might have proper types we can use from it everywhere 🤞 for when we migrate from the legacy to the new one.